File: front-end/types/notification.ts
// Tipos para o sistema de notificações

export type NotificationType =
  | 'process_started'
  | 'process_completed'
  | 'process_error'
  | 'review_approved'
  | 'review_rejected'
  | 'system_message'

export type NotificationVariant = 'default' | 'success' | 'error' | 'warning'

export interface Notification {
  id: string
  type: NotificationType
  title: string
  message: string
  timestamp: Date
  read: boolean
  actionUrl?: string  // Para navegação direta
  actionLabel?: string  // Ex: "Ver Resultados", "Baixar PDF"
  metadata?: {
    processId?: string
    batchId?: string
    documentId?: string
    cpf?: string
    reviewerEmail?: string
    documentCount?: number
    status?: string
    approvalReason?: string
  }
  variant?: NotificationVariant
}

// Tipo auxiliar para criar novas notificações (sem campos auto-gerados)
export type CreateNotificationInput = Omit<Notification, 'id' | 'timestamp'>

// Preferências para UI do centro de notificações
export interface NotificationPreferences {
  lastOpenedAt: Date | null
  autoMarkAsReadOnClick: boolean
}

File: back-end/app/models/notification.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime

class Notification(BaseModel):
    id: Optional[str] = None
    clinic_id: Optional[str] = None
    document_id: Optional[str] = None
    type: str
    title: str
    message: str
    variant: Optional[str] = None
    action_url: Optional[str] = None
    action_label: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None
    read: bool = False
    created_at: Optional[datetime] = None

class NotificationCreate(BaseModel):
    clinic_id: Optional[str] = None
    document_id: Optional[str] = None
    type: str
    title: str
    message: str
    variant: Optional[str] = None
    action_url: Optional[str] = None
    action_label: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None

class NotificationUpdate(BaseModel):
    read: Optional[bool] = None
